src/Commands/InstallCommand.php

```diff
@@ -3,22 +3,277 @@
 namespace Rapidez\Core\Commands;
 
 use Illuminate\Console\Command;
+use Illuminate\Support\Collection;
+
+use function Laravel\Prompts\confirm;
+use function Laravel\Prompts\multiselect;
+use function Laravel\Prompts\pause;
+use function Laravel\Prompts\select;
 
 class InstallCommand extends Command
 {
-    protected $signature = 'rapidez:install';
+    protected $signature = 'rapidez:install {--frontendonly : Just copy over the frontend files}';
 
     protected $description = 'Install Rapidez';
 
+    protected Collection $selectedPackages;
+
+    protected bool $dockerInstall = false;
+
     public function handle()
     {
-        copy(__DIR__ . '/../../package.json', base_path('package.json'));
-        copy(__DIR__ . '/../../postcss.config.js', base_path('postcss.config.js'));
-        copy(__DIR__ . '/../../tailwind.config.js', base_path('tailwind.config.js'));
-        copy(__DIR__ . '/../../vite.config.js', base_path('vite.config.js'));
-        copy(__DIR__ . '/../../yarn.lock', base_path('yarn.lock'));
-        copy(__DIR__ . '/../../.prettierrc.js', base_path('.prettierrc.js'));
-
-        $this->info('Done 🚀');
+        if ($this->option('frontendonly')) {
+            $this->frontend(true);
+
+            return;
+        }
+
+        $this
+            ->intro()
+            ->magento()
+            ->validate()
+            ->frontend()
+            ->packages()
+            ->finalize();
+    }
+
+    protected function intro()
+    {
+        $this->info(str_repeat('=', 37));
+        $this->info('  ____             _     _          ');
+        $this->info(" |  _ \ __ _ _ __ (_) __| | ___ ____");
+        $this->info(" | |_) / _` | '_ \| |/ _` |/ _ \_  /");
+        $this->info(' |  _ < (_| | |_) | | (_| |  __// / ');
+        $this->info(" |_| \_\__,_| .__/|_|\__,_|\___/___|");
+        $this->info('            |_|                     ');
+        $this->info('  Welcome to the Rapidez installer!');
+        $this->info(str_repeat('=', 37));
+        $this->newLine();
+        $this->warn('> You will get some questions, if you do not know what to choose;');
+        $this->warn('> Just enter through, the defaults are a good start.');
+        $this->newLine(2);
+
+        return $this;
+    }
+
+    protected function magento()
+    {
+        if (! confirm('Do you have a Magento installation up-and-running?')) {
+            if (confirm(
+                label: 'Shall we setup a demo Magento installation for you in Docker?',
+                hint: 'This will run `docker-compose up -d` and takes a while, make sure Docker is running!'
+            )) {
+                $this->dockerInstall = true;
+                if (passthru('docker-compose up -d') === false) {
```

Comment:
We should check wether all ports we want to use are available:
https://github.com/rapidez/rapidez/blob/55a324713fc860916e9b725cf8e19b8d9f930b1a/docker-compose.yml#L13
`TRAEFIK_PORT=80`

https://github.com/rapidez/rapidez/blob/55a324713fc860916e9b725cf8e19b8d9f930b1a/docker-compose.yml#L58
`DB_PORT=3306`

And if they are not, suggest or set a different port.

See: https://github.com/rapidez/rapidez/pull/82 for having domain resolution and the TRAEFIK_PORT variable available to set out of the box

Reply:
Follow-up on this with internal reference: RAP-925